src/tests/quantuminspire/qiskit/test_backend_qx.py

```diff
@@ -206,11 +206,17 @@ def test_get_experiment_results_returns_single_shot(self):
         job = QIJob('backend', '42', api)
         simulator = QuantumInspireBackend(api, Mock())
         experiment_result = simulator.get_experiment_results(job)[0]
```

Comment:
would be nice to have test that runs this few thousand times, with a bit more complex histogram, to see if the probability is similar.

Say, if the histogram is ` {'0': 0.2, '1': 0.3, '2': 0.4, '3': 0.1}` and we run this 10000 times, then the output of that test should be something like: ` {'0x2': 4015, '0x1': 2989, '0x3': 985, '0x0': 2011})`

Reply:
Added test_get_experiment_results_multiple_single_shots which tests your suggestion